type.go

```diff
@@ -374,6 +375,24 @@ func TypeID(typ Type) int {
 	return typ.ID()
 }
 
+// CanonicalUnionOfTypes returns the set of unique Types in types in sorted
+// order. types will be sorted in place, but a new slice of unique types will
+// be returned.
+func CanonicalUnionOfTypes(types []Type) []Type {
```

Comment:
Maybe call this `UniqueTypes`?

Reply:
Doesn't quite capture that its also sorted, but I like the simpler name.